protocol/message_persistence.go

```diff
@@ -198,12 +201,11 @@ func (db sqlitePersistence) tableUserMessagesScanAllFields(row scanner, message
 
 	if quotedText.Valid {
```

Comment:
If the `quotedText.Valid` is false because the text field is null the `common.QuotedMessage` would not be instantiated.

Reply:
oh, you are right, completely missed that

Reply:
wait, I think it can still crash.
So we pull from the db. If `quotedText` is not valid, `message.QuotedMessage` is `nil`.
But then, we call `PrepareImageURL` regardless, which in turns calls `m.QuotedMessage.PrepareImageURL`, in messenger

Reply:
Tests fail with a panic as a matter of fact
```
[2022-03-16T16:48:38.577Z] 2022-03-16T16:48:09.051Z	DEBUG	protocol/messenger.go:1326	running shutdown task	{"n": 9}
[2022-03-16T16:48:38.577Z]     suite.go:63: test panicked: runtime error: invalid memory address or nil pointer dereference
[2022-03-16T16:48:38.577Z]         goroutine 16298 [running]:
[2022-03-16T16:48:38.577Z]         runtime/debug.Stack()
[2022-03-16T16:48:38.577Z]         	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
[2022-03-16T16:48:38.577Z]         github.com/stretchr/testify/suite.failOnPanic(0xc000c21520)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:63 +0x3e
[2022-03-16T16:48:38.577Z]         panic({0x1459920, 0x23ee110})
[2022-03-16T16:48:38.577Z]         	/usr/local/go/src/runtime/panic.go:1038 +0x215
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol/common.(*QuotedMessage).PrepareImageURL(...)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/common/message.go:38
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol/common.(*Message).PrepareImageURL(0xc000ce0900, 0xc)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/common/message.go:165 +0x5e
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*Messenger).prepareMessages(0xc0009ea540, 0x161b45f)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/messenger.go:3820 +0x99
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*Messenger).sendChatMessage(0xc0009ea540, {0x1985370, 0xc0000440d8}, 0xc00028f200)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/messenger.go:2420 +0xe25
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*Messenger).SendChatMessage(...)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/messenger.go:2273
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*Messenger).SendChatMessages(0xc000b9c910, {0x1985370, 0xc0000440d8}, {0xc000130c98, 0x1, 0x1})
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/messenger.go:2281 +0x85
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*Messenger).InviteUsersToCommunity(0xc0009ea540, 0xc000717610)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/messenger_communities.go:655 +0x1c5
[2022-03-16T16:48:38.577Z]         github.com/status-im/status-go/protocol.(*MessengerCommunitiesSuite).TestBanUser(0xc00040b440)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/src/github.com/status-im/status-go/protocol/communities_messenger_test.go:1076 +0x41f
[2022-03-16T16:48:38.577Z]         reflect.Value.call({0xc0008f5080, 0xc000130608, 0x0}, {0x160ae56, 0x4}, {0xc000717e70, 0x1, 0xc000717d80})
[2022-03-16T16:48:38.577Z]         	/usr/local/go/src/reflect/value.go:543 +0x814
[2022-03-16T16:48:38.577Z]         reflect.Value.Call({0xc0008f5080, 0xc000130608, 0xc00040b440}, {0xc000717e70, 0x1, 0x1})
[2022-03-16T16:48:38.577Z]         	/usr/local/go/src/reflect/value.go:339 +0xc5
[2022-03-16T16:48:38.577Z]         github.com/stretchr/testify/suite.Run.func1(0xc000c21520)
[2022-03-16T16:48:38.577Z]         	/home/jenkins/workspace/status-go_prs_tests_PR-2591/pkg/mod/github.com/stretchr/testify@v1.7.0/suite/suite.go:158 +0x4b6
[2022-03-16T16:48:38.577Z]         testing.tRunner(0xc000c21520, 0xc0004f5710)
[2022-03-16T16:48:38.577Z]         	/usr/local/go/src/testing/testing.go:1259 +0x102
[2022-03-16T16:48:38.577Z]         created by testing.(*T).Run
```

So I think @flexsurfer might not be running this PR in local dev, it would crash on the first received message